File: app/src/main/java/ru/mamykin/foreignbooksreader/di/modules/DatabaseModule.java
package ru.mamykin.foreignbooksreader.di.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import ru.mamykin.foreignbooksreader.database.BookDao;
import ru.mamykin.foreignbooksreader.database.BookDatabaseHelper;

/**
 * Creation date: 5/29/2017
 * Creation time: 11:39 AM
 * @author Andrey Mamykin(mamykin_av)
 */
@Module
public class DatabaseModule {
    @Provides
    @Singleton
    BookDatabaseHelper provideBooksDatabaseHelper(Context context) {
        return new BookDatabaseHelper(context);
    }

    @Provides
    @Singleton
    BookDao provideBookDao(BookDatabaseHelper dbHelper) {
        return dbHelper.getBookDao();
    }
}

File: app/src/main/java/ru/mamykin/foboreader/di/modules/DatabaseModule.kt
package ru.mamykin.foboreader.di.modules

import android.content.Context
import dagger.Module
import dagger.Provides
import ru.mamykin.foboreader.data.database.BookDao
import ru.mamykin.foboreader.data.database.BookDatabaseHelper
import javax.inject.Singleton

@Module
@Singleton
class DatabaseModule {

    @Provides
    @Singleton
    fun provideBooksDatabaseHelper(context: Context): BookDatabaseHelper {
        return BookDatabaseHelper(context)
    }

    @Provides
    @Singleton
    fun provideBookDao(dbHelper: BookDatabaseHelper): BookDao {
        return dbHelper.getBookDao()
    }
}